AEPCore/Sources/eventhub/ExtensionContainer.swift

```diff
@@ -96,6 +96,14 @@ extension ExtensionContainer: ExtensionRuntime {
         return EventHub.shared.getSharedState(extensionName: extensionName, event: event, barrier: barrier)
     }
 
+    func createSharedState(data: [String: Any], xdmData: [String: Any], event: Event?) {
```

Comment:
should we do it similar to https://github.com/adobe/aepsdk-core-ios/pull/437/files#diff-02bd1581c751b39213c003d305212478496f8a5290a1a877cef358a338c892f4R158, make xdmData optional and only keep one `createSharedState ` method ？

Reply:
I opted to overload it for a few reasons
1. Provide an API to create shared state without needing to interact/understand what XDM data is
2. Reduce breaking changes/updates in other extensions by updating this API

thoughts?

Reply:
either way, the extension developers need to know whether they need to deal with XDM data (by selecting which method to use vs using an optional parameter)
if we don't mind breaking the current extensions and do an update to all, actually I'd prefer to keep this new method to force them passing into both data even if one of them is empty.

Reply:
updated.